cmake/Modules/FindIrrlicht.cmake

```diff
@@ -29,8 +29,22 @@ foreach(libname IN ITEMS IrrlichtMt Irrlicht)
 	endif()
 endforeach()
 
-# Users will likely need to edit these
-mark_as_advanced(CLEAR IRRLICHT_LIBRARY IRRLICHT_INCLUDE_DIR)
+# Handholding for users
+if(IRRLICHT_INCLUDE_DIR AND (NOT IS_DIRECTORY "${IRRLICHT_INCLUDE_DIR}" OR
+	NOT EXISTS "${IRRLICHT_INCLUDE_DIR}/irrlicht.h"))
+	message(WARNING "IRRLICHT_INCLUDE_DIR was set to ${IRRLICHT_INCLUDE_DIR} "
+		"but irrlicht.h does not exist inside. The path will not be used.")
+	unset(IRRLICHT_INCLUDE_DIR CACHE)
+endif()
+if(WIN32 OR CMAKE_SYSTEM_NAME STREQUAL "Linux" OR APPLE)
```

Comment:
How about *BSD?

Reply:
Should be covered by `.a ` and `.so` but I didn't care to find out exactly.